Examples: Update example to restore 6.4 auto-title behavior in UI

shilman opened the pull request:
Issue: N/A

## What I did

- [x] Update `MIGRATION.md` with instructions for restoring old behavior
- [x] Update `examples/react-ts` with working example

@tmeasday Any suggestions for how to make this snippet forward-compatible with 7.0?

## How to test

```
yarn bootstrap --core
cd examples/react-ts
yarn storybook
# remove manager.js & re-run to see difference
```

shilman committed:
Examples: Update example to restore 6.4 auto-title behavior in UI

tmeasday reviewed (approved):
Lgtm

tmeasday commented:
I think the snippet is forward compatible.. why wouldn't it be?

shilman commented:
@tmeasday I thought we were going to remove `isLeaf` in 7.0 and replace it with `type`?

tmeasday commented:
Oh right, I missed that detail. 

Yes you would want to check `type === 'story'`. Perhaps we should add that now so people don't use a old api now.

tmeasday commented:
Should we keep isLeaf and co in 7.0 for back compat?

shilman commented:
@tmeasday Let's add the `type` field now to make the snippet forward-compatible.

tmeasday committed:
Add `type` annotation to `StoriesHash`

For forwards compatibility with changes coming in 7.0, see: https://github.com/storybookjs/storybook/pull/18023

tmeasday commented:
@shilman done. Shall I update the example too?

shilman committed:
Examples: Update auto-title example to use forward-compatible API
